README.md

```diff
@@ -1,85 +1,73 @@
 
-Apply the storage class
+### VolumeSnapshotClassBackupItemAction
 
-```
-kubectl apply -f examples/sc.yaml
-```
+A BackupItemAction plugin that backs up [`snapshot.storage.k8s.io.volumesnapshotclasses`][5].
 
-Apply the snapshot class
+This plugin will look for snapshot list operation secret from the [annotations][6] on the VolumeSnapshotClass object being backed up.
 
-```
-kubectl apply -f examples/snapshotclass.yaml
-```
+### PVCRestoreItemAction
 
-Create the demo application
+A RestoreItemAction plugin that restores `persistentvolumeclaims` that were backed up by [PVCBackupItemAction](#PVCBackupItemAction).
 
-```
-kubectl apply -f examples/demo.yaml
-```
+This plugin will modify the spec of the `persistentvolumeclaim` being restored to use the VolumeSnapshot, created during backup, as the data source ensuring that the newly provisioned volume, to satisfy this claim, may be pre-popuated using the VolumeSnapshot.
 
-Either start Velero locally, or edit a deployment, and provide the following flag to update the order in which to restore resources:
+### VolumeSnapshotRestoreItemAction
 
-```
---restore-resource-priorities namespaces,storageclasses,customresourcedefinitions,volumesnapshotclass.snapshot.storage.k8s.io,volumesnapshots.snapshot.storage.k8s.io,volumesnapshotcontents.snapshot.storage.k8s.io,persistentvolumes,persistentvolumeclaims,secrets,configmaps,serviceaccounts,limitranges,pods,replicaset
-```
+A RestoreItemAction plugin that restores [`volumesnapshots.snapshot.storage.k8s.io`][3]. 
 
-Create a backup, including cluster resources so that the VolumeSnapshotContents objects are properly backed up (walking from a VolumeSnapshot to the associated Contents will be addressed at a later time):
+This plugin will use the annotations, added during backup, to create a [`volumesnapshotcontent.snapshot.storage.k8s.io`][4] and statically bind it to the volumesnapshot object being restored. The plugin will also set the necessary [annotations][6] if the original volumesnapshotcontent had snapshot deletion secrets associated with it. 
 
-```
-velero backup create --include-cluster-resources=true --include-namespaces demo demo-backup --wait
-```
+### VolumeSnapshotClassRestoreItemAction
 
-Simulate a disaster:
+A RestoreItemAction plugin that restores [`snapshot.storage.k8s.io.volumesnapshotclasses`][5]. 
 
-```
-kubectl delete ns/demo
-```
+This plugin will use the [annotations][6] on the object being restored to return, as additional items, any snapshot lister secret that is associated with the volumesnapshotclass.
 
-Recover
+## Building the plugins
 
+To build the plugins, run
+
+```bash
+$ make
 ```
-velero restore create --from-backup demo-backup
-```
+
+## Known shortcomings
+
+We are tracking known limitations with the plugins [here][2]
+
+[1]: https://github.com/vmware-tanzu/velero/releases
+[2]: https://velero.io/docs/v1.4.0/csi/
```

Comment:
This is a future link.

Reply:
this will probably end up being "https://velero.io/docs/v1.4/csi/" since we're going to do a docs only for each major.minor combo.

Another thing we can do is add a shortlink here: https://github.com/vmware-tanzu/velero/blob/master/site/_data/shortlinks.yml

Which then allows you to use a link like "velero.io/docs/csi" which will automatically redirect to the latest version. That's probably the way to go here.